File: InsertLocalDbNoTransaction/app/src/main/java/com/algokelvin/insertdb/db/dao/UserDao.java
package com.algokelvin.insertdb.db.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.algokelvin.insertdb.db.entity.User;

import java.util.List;

@Dao
public interface UserDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(List<User> user);

    @Query("SELECT * FROM users")
    List<User> getAllUsers();
}

File: ECommerce/app/src/main/java/com/algokelvin/shoppingyuk/data/db/UserDao.kt
package com.algokelvin.shoppingyuk.data.db

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.algokelvin.shoppingyuk.data.model.user.User

@Dao
interface UserDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun saveUser(user: User)

    @Query("SELECT * FROM users_data WHERE id = :id")
    suspend fun getUser(id: Int): User
}